pytorch/inference/docker/2.2/py3/Dockerfile.graviton.cpu

```diff
@@ -0,0 +1,253 @@
+#                                         |___/
+#  ____           _
+# |  _ \ ___  ___(_)_ __   ___ 
+# | |_) / _ \/ __| | '_ \ / _ \
+# |  _ <  __/ (__| | |_) |  __/
+# |_| \_\___|\___|_| .__/ \___|
+#                  |_|
+########################################################
+FROM arm64v8/ubuntu:20.04 as ec2
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+ARG PYTHON
+ARG PYTHON_VERSION
+ARG OPEN_MPI_VERSION
+ARG MAMBA_VERSION
+
+# PyTorch Passed arguments (URLs and Versions)
+ARG TORCH_URL
+ARG TORCHVISION_URL
+ARG TORCHAUDIO_URL
+ARG TORCHTEXT_URL
+ARG TORCHSERVE_VERSION
+
+### Common ENV Variables
+ENV DEBIAN_FRONTEND=noninteractive \
+    LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib" \
+    LANG=C.UTF-8 \
+    LD_LIBRARY_PATH=/opt/conda/lib/:$LD_LIBRARY_PATH \
+    PATH=/opt/conda/bin:$PATH \
+    TEMP=/home/model-server/tmp \
+    ### Graviton Optimization ENV Variables
+    LRU_CACHE_CAPACITY=1024 \
+    THP_MEM_ALLOC_ENABLE=1 \
+    DNNL_DEFAULT_FPMATH_MODE=BF16
+
+RUN apt-get update \
+ && apt-get -y upgrade \
+ && apt-get install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    cmake \
+    curl \
+    emacs \
+    # adding GCC 10 for graviton compile support
+    gfortran-10 \
+    g++-10 \
+    gcc-10 \
+    git \
+    jq \
+    libgl1-mesa-glx \
+    libglib2.0-0 \
+    libsm6 \
+    libxext6 \
+    libxrender-dev \
+    openjdk-17-jdk \
+    openssl \
+    scons \
+    vim \
+    wget \
+    unzip \
+    zlib1g-dev \
+    libjpeg-dev \
+    libpng-dev \
+ && apt-get autoremove -y \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+ # Set default gcc, python and pip versions
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 1 && \
+    update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 1 && \
+    update-alternatives --install /usr/bin/gfortran gfortran /usr/bin/gfortran-10 1
+
+# Install OpenMPI
+RUN wget https://www.open-mpi.org/software/ompi/v4.1/downloads/openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+# The ENV variables declared below are changed in the previous section
+# Grouping these ENV variables in the first section causes
+# ompi_info to fail. This is only observed in CPU containers
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value
+
+# Install CondaForge miniconda
+RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-aarch64.sh \
+ && chmod +x ~/mambaforge.sh \
+ && ~/mambaforge.sh -b -p /opt/conda \
+ && rm ~/mambaforge.sh \
```

Comment:
If you do end up upgrading mamba version, I'd recommend updating the url to use miniforge3 instead. As of 9/2023, mambaforge is discouraged because there are now no diff between mambaforge and miniforge apart from the name so they decided to merge the two packages into miniforge. Miniforge3 is just for py310 while miniforge is just for py39. You can read more [here](https://github.com/conda-forge/miniforge).

You can get the releases url link from [here](https://github.com/conda-forge/miniforge/releases).

You can also copy the installation [here](https://github.com/aws/deep-learning-containers/blob/f237519a5e6f1dcf9204610bab167f8417e235ca/pytorch/inference/docker/2.2/py3/cu118/Dockerfile.gpu#L172-L176).

Reply:
done